src/shmessy/__init__.py

```diff
@@ -36,21 +33,35 @@ def infer_schema(self, df: DataFrame) -> ShmessySchema:
         return ShmessySchema(columns=columns, infer_duration_ms=infer_duration_ms)
 
     def fix_schema(
-        self, df: DataFrame, fix_column_names: Optional[bool] = False
+        self, df: DataFrame, *, fix_column_names: Optional[bool] = False
     ) -> DataFrame:
         for column in df:
             df[column] = self.__validators_handler.fix_field(
                 column=df[column], sample_size=self.__sample_size
             )
 
         if fix_column_names:
-            df = self._fix_column_names(df)
+            df = _fix_column_names(df)
 
         return df
 
-    @staticmethod
-    def _fix_column_names(df: DataFrame) -> DataFrame:
-        fixed_column_names = {}
-        for column in df.columns:
-            fixed_column_names[column] = re.sub("[^0-9a-zA-Z]+", "_", column)
-        return df.rename(columns=fixed_column_names)
+    def read_csv(
+        self,
+        filepath_or_buffer: Union[str, TextIO, BinaryIO],
+        *,
+        use_sniffer: Optional[bool] = True
+    ) -> DataFrame:
+
+        if use_sniffer:
+            dialect = csv.Sniffer().sniff(
```

Comment:
if sniffer fails? there is no error handling here?

Reply:
If the sniffer cant identify some values it retuns his default values...
Any other issue will be raise automatically.